src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -95,14 +95,10 @@ impl Res {
         }
     }
 
-    fn def_id(self) -> DefId {
-        self.opt_def_id().expect("called def_id() on a primitive")
-    }
-
-    fn opt_def_id(self) -> Option<DefId> {
+    fn def_id(self, tcx: TyCtxt<'_>) -> DefId {
         match self {
-            Res::Def(_, id) => Some(id),
-            Res::Primitive(_) => None,
+            Res::Def(_, id) => id,
+            Res::Primitive(prim) => *PrimitiveType::primitive_locations(tcx).get(&prim).unwrap(),
```

Comment:
Are we sure we always have all the primitives? I guess unless new primitives are added we're fine, but I'm still feeling a bit uneasy... Well, not a blocker in any case.

Reply:
I can go through them one by one this afternoon, but they should all be there now that primitives are documented in core. This will cause an ICE if you use no_core and try to link to a primitive, but no_core isn't really supported anyway.